src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -133,10 +126,14 @@ public JsonWebKey supplyKey(
         return keysByKid.get(kid);
     }
 
-    private String[] splitScopes(
-            String scopes)
+    private OAuthRealm newOAuthRealm(
+        String realmName)
     {
-        return scopes.split("\\s+");
+        if (nextRealmBitShift == MAX_REALMS)
+        {
+            throw new IllegalStateException("Too many realms");
+        }
+        return new OAuthRealm(realmName, 1L << nextRealmBitShift++ << MAX_SCOPES);
```

Comment:
Recommend changing the `OAuthRealm` constructor parameter `realmId` to `realmBitShift`, then pass from here as `nextRealmBitShift++`, and in the constructor perform the `realmId = 1L << realmBitShift << MAX_SCOPES`, keeping `MAX_SCOPES` defined as a private constant inside `OAuthRealm`, not in `OAuthRealms`.